### PR TITLE
Adding GCN and GIN

### PR DESCRIPTION
**Pull request by rahulsk88**
Added GCN and GIN along with Cell_Line Featurizer.

### CODE REVIEW

File: src/models/Cell_line.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+
+import torch
+import torch.nn as nn
```
Comment (davidanugraha) [inline src/models/Cell_line.py:3]: Could you please move this Cell_line model to the generic_model.py? I have another Cell_line model there; maybe mention as alternative if it's entirely different

File: src/models/GCN.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
+import torch
+import torch.nn as nn
+from torch_geometric.nn import GCNConv, global_max_pool
+from .Cell_line import *
```
Comment (davidanugraha) [inline src/models/GCN.py:4]: Similarly with all other files, please fix naming convention into pot_hole format

File: src/models/GIN.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+    ## Dense concatted layers
+    x_all = self.relu(self.fc1_mlp(x_all))
+    x_all = self.dropout(x_all)
+    x_all = self.relu(self.fc2_mlp(x_all))
+    x_all = self.dropout(x_all)
+    output = self.fc3_mlp(x_all)
+
+    ## Limiting output to -100 to 100
+    sigmoid = torch.nn.Sigmoid()
```
Comment (davidanugraha) [inline src/models/GIN.py:167]: I was doing similar to this, but then I realized nn.Sigmoid might favor input that is closer to the center and disfavoring results that are close to 100 or -100, i.e. we sort of assuming such as our distribution. I used torch.clamp instead. We should discuss this though.

File: src/models/GIN.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        elif isinstance(m, GINConv): 
+            for sub_module in m.nn.modules(): ## GIN networks have Linear Layers inside of them  So might need to double check.
+                if isinstance(sub_module, nn.Linear):
+                    nn.init.kaiming_uniform_(sub_module.weight, nonlinearity='relu')
+                if sub_module.bias is not None:
+                    nn.init.constant_(sub_module.bias, 0)
+
+
+  def forward(self, data):
+    x_d1, edge_index_d1, cellline, batch = data.x1, data.edge_index1, data.batch, data.cellline
```
Comment (davidanugraha) [inline src/models/GIN.py:115]: Current DataLoader does not have data.batch actually

File: src/models/GCN.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+                loss.backward()
+                self.optimizer.step()
+                epoch_loss += loss.item()
+
+            self.scheduler.step() 
+
+            avg_train_loss = epoch_loss / len(train_loader)
+            val_loss = self.validate(val_loader, device)
+
+            print(f'Epoch {epoch+1}/{epochs}, Train Loss: {avg_train_loss:.4f}, Val Loss: {val_loss:.4f}, LR: {self.scheduler.get_last_lr()[0]}')
```
Comment (davidanugraha) [inline src/models/GCN.py:169]: Replace all prints with logging.info / logging.debug to be consistent

File: src/models/GIN.py (hunk 4 above)
Comment (davidanugraha) [reply src/models/GIN.py:115]: In addition, the dataloader does not support data.cellline

**Review by davidanugraha — CHANGES_REQUESTED** [review]
Please see the comments below (I have reviewed everything, just took a glance, but I don't want to overwhelm), also please add the model to main.py and try to run it, see if it runs (just test 1 epoch or something).
